### DISCUSSION TITLE
Excessive render of toggled icon

### DISCUSSION THREAD

**Discussion by MarioPerini** [alpinejs/alpine#2351] [7. Bugs (archived)]
I use Fontawesome Icons as SVG's and have a performance issue when using it combined with alpine.

Here is a simple example:
https://codepen.io/MarioPerini/pen/ZEJjYzj

If you look at chrome-dev console you will see a constant re-render of the parent and the icon svg.
![image](https://user-images.githubusercontent.com/40631006/141121756-8369dd01-d63d-44d7-8fc2-84abac74edfd.png)

This causes a significant load on CPU and GPU. I opened the tab about half-time of the chart:
![image](https://user-images.githubusercontent.com/40631006/141122423-8a4056c5-9e58-4dff-a823-44fc7e8d6454.png)

Expected behaviour:
Render once and re-render only on click

**Comment by SimoTod** [comment]
I think you can use x-ignore on the svg so alpine doesn't try to crawl its children

**Reply by MarioPerini** [reply]
You can't. it's a 
`<i class="fa fa-something"></i>`
And will be replaced by fontawesome with an svg.

**Reply by philwolstenholme** [reply]
Have you checked the attribute idea? https://fontawesome.com/v5.15/how-to-use/on-the-web/other-topics/accessibility suggests that a `title` and `aria-hidden` attributes that you add to the `i` element will also be added to the `svg` element, so maybe you could do `<i class="fa fa-something" x-ignore></i>` or perhaps try `<i class="fa fa-something" x-ignore="true"></i>`.

**Reply by MarioPerini** [reply]
Yes i did check that before i opened the discussion. Just updated the codepen: https://codepen.io/MarioPerini/pen/ZEJjYzj
The issue is still present.

**Comment by danddanddand** [comment]
I feel like there could be something wrong with your set up of Font Awesome. We use it extensively with Alpine and have never had performance issue. I do tend to treat the `<i class="fas fa-something"></i>` tags more like a character, so if we need to do styling or conditionally render/show/hide with x-if or x-show it would probably be on span that wraps it. You can inspect [www.borderconnect.com](https://www.borderconnect.com/) source and see a bunch of examples of how it's used on our site.

**Reply by danddanddand** [reply]
I played with example you had up and wrapping the `<i></i>` tags in span for alpine is alternative (https://codepen.io/danddanddand/pen/MWvBpBL). It's bit more verbose unfortunately.

**Reply by MarioPerini** [reply]
I will probably use this as workaround. but i still think it is a bug.
